supplementary/alica_ros1/alica_tests/src/test/test_blackboard.cpp

```diff
@@ -117,5 +145,452 @@ TEST_F(TestBlackBoard, testMapping)
     EXPECT_EQ(targetLocked.get<int64_t>("valueInt"), 1);
 }
 
+TEST_F(TestBlackboard, testMappingFromBool)
+{
+    Blackboard srcBb, targetBb;
+    auto srcLocked = LockedBlackboardRW(srcBb);
+    auto targetLocked = LockedBlackboardRW(targetBb);
+
+    srcLocked.set<bool>("valueSrc", true);
+    targetLocked.set<bool>("valueTarget", false);
+    EXPECT_TRUE((checkMapping<bool>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", true)));
+
+    targetLocked.set<int64_t>("valueTarget", -1);
+    EXPECT_TRUE((checkMapping<int64_t>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", 1)));
+
+    targetLocked.set<double>("valueTarget", -1.0);
+    EXPECT_TRUE((checkMapping<double>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", 1.0)));
+
+    targetLocked.set<std::string>("valueTarget", "");
+    EXPECT_THROW(({ checkMapping<std::string>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", ""); }), BlackboardException);
+
+    srcLocked.set<bool>("valueSrc", false);
+    targetLocked.set<bool>("valueTarget", true);
+    EXPECT_TRUE((checkMapping<bool>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", false)));
+
+    targetLocked.set<int64_t>("valueTarget", 1);
+    EXPECT_TRUE((checkMapping<int64_t>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", 0)));
+
+    targetLocked.set<double>("valueTarget", 1.0);
+    EXPECT_TRUE((checkMapping<double>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", 0)));
+
+    targetLocked.set<std::string>("valueTarget", "");
+    EXPECT_THROW(({ checkMapping<std::string>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", ""); }), BlackboardException);
+}
+
+TEST_F(TestBlackboard, testMappingFromInt64)
+{
+    Blackboard srcBb, targetBb;
+    auto srcLocked = LockedBlackboardRW(srcBb);
+    auto targetLocked = LockedBlackboardRW(targetBb);
+
+    srcLocked.set<int64_t>("valueSrc", 1);
+    targetLocked.set<bool>("valueTarget", false);
+    EXPECT_TRUE((checkMapping<bool>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", true)));
+
+    targetLocked.set<int64_t>("valueTarget", 0);
+    EXPECT_TRUE((checkMapping<int64_t>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", 1)));
+
+    targetLocked.set<double>("valueTarget", 0.0);
+    EXPECT_TRUE((checkMapping<double>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", 1.0)));
+
+    targetLocked.set<std::string>("valueTarget", "");
+    EXPECT_THROW(({ checkMapping<std::string>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", ""); }), BlackboardException);
+
+    srcLocked.set<int64_t>("valueSrc", -1);
+    targetLocked.set<bool>("valueTarget", false);
+    EXPECT_TRUE((checkMapping<bool>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", true)));
+
+    targetLocked.set<int64_t>("valueTarget", 0);
+    EXPECT_TRUE((checkMapping<int64_t>(srcBb.impl(), targetBb.impl(), "valueSrc", "valueTarget", -1)));
+
+    targetLocked.set<double>("valueTarget", 0.0);
```

Comment:
as I had said map is an internal method used to map keys that are set in the pml file. You have instead used set to influence the type of the target. That is relying on internal behaviour. Anyway I'll fix it